src/proto_nd_flow/reco/charge/calib_prompt_hits.py

```diff
@@ -152,14 +155,18 @@ def run(self, source_name, source_slice, cache):
             mask = (packets_data['packet_type'] == 0) & mask
             n = np.count_nonzero(mask)
             packets_arr = packets_data.data[mask]
-            packet_frac_bt_arr = packet_frac_bt.data[mask]
-            packet_seg_bt_arr = packet_seg_bt.data[mask]
+            if resources['RunData'].is_mc:
+                packet_frac_bt_arr = packet_frac_bt.data[mask]
+                packet_seg_bt_arr = packet_seg_bt.data[mask]
             index_arr = packets_index.data[mask]
         else:
             n = 0
             index_arr = np.zeros((0,), dtype=packets_index.dtype)
 
-        has_mc_truth = packet_seg_bt is not None
+        if resources['RunData'].is_mc:
+            has_mc_truth = packet_seg_bt is not None
+        else:
+            has_mc_truth = False
```

Comment:
At this point `has_mc_truth` is redundant with `resources['RunData'].is_mc`

Reply:
Is there no weird situation where it's MC but without backtracking packets?